src/modal.ts

```diff
@@ -0,0 +1,371 @@
+  infoCvv.classList.add('info__cvv');
+  const infoCvvValid = document.createElement('label');
+  infoCvvValid.classList.add('info__cvv-valid');
+  const cvvValidInput = document.createElement('input');
+  cvvValidInput.classList.add('input', 'cvv-valid__input');
+  cvvValidInput.setAttribute('type', 'text')
+  cvvValidInput.setAttribute('title', 'Введите месяц год (01/12 23/29)')
+  const infoCvvCvv = document.createElement('label');
+  infoCvvCvv.classList.add('info__cvv-cvv');
+  infoCvvCvv.setAttribute('type', 'number')
+  const cvvCvvInput = document.createElement('input');
+  cvvCvvInput.classList.add('input', 'cvv-cvv__input');
+  cvvCvvInput.setAttribute('type', 'number')
+  cvvCvvInput.setAttribute('title', 'Введите 3-и цифры')
+  const cvvValidSpan = document.createElement('span');
+  cvvValidSpan.classList.add('cvv-valid__span');
+  const cvvSvvSpan = document.createElement('span');
+  cvvSvvSpan.classList.add('cvv-cvv__span');
+  const cardNumberError = document.createElement('div');
+  cardNumberError.classList.add('card-number__error');
+  const cvvValidError = document.createElement('div');
+  cvvValidError.classList.add('cvv-valid__error');
+  const cvvCvvError = document.createElement('div');
+  cvvCvvError.classList.add('cvv-cvv__error');
+
+
+
+  main?.append(wrapperModal)
+  wrapperModal.append(wrapperModalForm)
+  wrapperModalForm.append(formModal)
+
+  formModal.append(modalDetails, modalDetailsName, modalDetailsPhone, modalDetailsAddress, modalDetailsEmail, modalDetailsCards, modalDetailsBtn)
+
+  modalDetailsName.append(inputName, modalDetailsNameTextError, modalDetailsNameText)
+  modalDetailsPhone.append(inputPhone,modalDetailsPhoneTextError, modalDetailsPhoneText)
+  modalDetailsAddress.append(inputAddress,modalDetailsAddressTextError, modalDetailsAddressText)
+  modalDetailsEmail.append(inputEmail,modalDetailsEmailTextError, modalDetailsEmailText)
+
+  modalDetailsCards.append(detailsCardsText, detailsCardsInfo, cardNumberError, cvvValidError, cvvCvvError)
+  detailsCardsInfo.append(infoCardNumber, infoCvv)
+
+  infoCardNumber.append(infoCardNumberInput)
+  infoCardNumberInput.append(cardNumberCardName, cardNumberNumber)
+
+  infoCvv.append(infoCvvValid, infoCvvCvv)
+  infoCvvValid.append(cvvValidSpan, cvvValidInput)
+  infoCvvCvv.append(cvvSvvSpan, cvvCvvInput)
+
+  modalDetails.innerHTML = 'Персональные данные';
+  modalDetailsNameText.innerHTML = 'Ваня Пупкин';
+  modalDetailsPhoneText.innerHTML = '+375291234545';
+  modalDetailsAddressText.innerHTML = 'Край Область Регион';
+  modalDetailsEmailText.innerHTML = 'admin@mail.ru';
+  detailsCardsText.innerHTML = 'Данные карты';
+  cvvValidSpan.innerHTML = 'VALID:';
+  cvvSvvSpan.innerHTML = 'CVV:';
+  modalDetailsBtn.innerHTML = 'Подтвердить';
+  validation();
+
+wrapperModal.addEventListener('click', (event) => {
+  if ((<HTMLElement>event.target).classList.contains('wrapper-modal')) {
+    wrapperModal.outerHTML = '';
+  }
+})
+}
+
+
+function validation() {
```

Comment:
function validation should not contain functional for creating any elements, only for validation